examples/typescript/swap.ts

```diff
@@ -0,0 +1,275 @@
+/**
+ * Example to demonstrate creating and adding to liquidity pools, swapping between two fungible asset.
+ *
+ * Steps:
+ * 1. Create two accounts (Alice and Bob)
+ * 2. Fund Alice and Bob
+ * 3. Create a FA with Alice as the owner
+ * 4. Create another FA with Bob as the owner
+ * 5. Create a liquidity pool with Alice's FA and Bob's FA
+ * 6. Swap between Alice's FA and Bob's FA
+ */
+
+import {
+  Account,
+  AccountAddress,
+  Aptos,
+  Bool,
+  Ed25519PrivateKey,
+  Hex,
+  HexInput,
+  MoveObject,
+  MoveString,
+  U64,
+  ViewRequestData,
+} from "aptos";
+
+const readline = require("readline").createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const getOptimalLpAmount = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  token1Addr: HexInput,
+  token2Addr: HexInput,
+): Promise<void> => {
+  const payload: ViewRequestData = {
+    function: `0x${swap.toStringWithoutPrefix()}::router::optimal_liquidity_amounts`,
```

Comment:
Good callout here for improvements to the SDK to remove the 0x in the pattern

Reply:
Not sure you mean. There's still a 0x at the start of the format string

Reply:
There's something we have to fix in the SDK, which adds the protection of some inputs as `0x${string}`, however if you don't type the output of `swap.toStringWithoutPrefix{}` to `0x${string}`, it won't accept it (even if it always outputs `0x12345`)

We tried something fancy, and it made it kinda messy

Reply:
Does the API not accept module addresses without 0x prefix?

Reply:
It does, but we're trying to get rid of the variability

Reply:
This is actually fixed now, so we can get rid of the 0x weird hack.